flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -310,26 +310,33 @@ public static CompletedCheckpointStore createCompletedCheckpointStore(
      *
      * @param resourceRequirements resource requirements in pod template
      * @param mem Memory in mb.
+     * @param memoryLimitFactor limit factor for the memory, used to set the limit resources.
      * @param cpu cpu.
+     * @param cpuLimitFactor limit factor for the cpu, used to set the limit resources.
      * @param externalResources external resources
      * @param externalResourceConfigKeys config keys of external resources
      * @return KubernetesResource requirements.
      */
     public static ResourceRequirements getResourceRequirements(
             ResourceRequirements resourceRequirements,
             int mem,
+            double memoryLimitFactor,
             double cpu,
+            double cpuLimitFactor,
             Map<String, ExternalResource> externalResources,
             Map<String, String> externalResourceConfigKeys) {
         final Quantity cpuQuantity = new Quantity(String.valueOf(cpu));
+        final Quantity cpuLimitQuantity = new Quantity(String.valueOf(cpu * cpuLimitFactor));
         final Quantity memQuantity = new Quantity(mem + Constants.RESOURCE_UNIT_MB);
+        final Quantity memQuantityLimit =
+                new Quantity(((int) (mem * memoryLimitFactor)) + Constants.RESOURCE_UNIT_MB);
```

Comment:
Maybe we do not need to use the `int` here since K8s could also support decimal amount for memory resource. e.g. `1024.5m` `1.5g`. But the current implementation also make sense to me.